common/fixture/atxs.go

```diff
@@ -77,3 +77,21 @@ func ToAtx(t testing.TB, watx *wire.ActivationTxV1) *types.ActivationTx {
 	atx.TickCount = 1
 	return atx
 }
+
+type idMatcher types.ATXID
+
+func MatchId(id types.ATXID) idMatcher {
+	return idMatcher(id)
+}
+
+func (m idMatcher) Matches(x any) bool {
+	type hasID interface {
+		ID() types.ATXID
+	}
+	v, ok := x.(hasID)
+	return ok && v.ID() == types.ATXID(m)
+}
+
+func (m idMatcher) String() string {
+	return "is ATX ID " + types.ATXID(m).String()
+}
```

Comment:
What is this for? this doesn't seem to be used anywhere?

Reply:
but why? It's a one liner that shows clearly what it does, why hide this inside a different type in a different package?

Reply:
Technically, it's a 3-liner :). IMHO, `fixture.MatchId(atx.ID())` is a self-explainable one-liner. The
```go
gomock.Cond(func(atx any) bool {
	return atx.(*types.ActivationTx).ID() == watx.ID()
})
```
is convoluted and brings me a headache every time I need to write it. It's also the reason why there are many:
```go
atxHdlr.mbeacon.EXPECT().OnAtx(gomock.Any())
```
in the code.

Reply:
I don't think that's the reason why, but rather that when the `gomock.Any()` version was written the writer didn't know that there was an option to only assert the ID 😉 

> Technically, it's a 3-liner :)

Only because I added line breaks that aren't needed 😉

Reply:
How shall we proceed with this?

Reply:
Removed